runelite-client/src/main/java/net/runelite/client/plugins/specorb/SpecOrbOverlay.java

```diff
@@ -57,20 +64,28 @@
 	private static final int RUN_ORB_Y_RESIZABLE = 103;
 
 	private static final Color SPECIAL_ORB_BACKGROUND_COLOR = new Color(51, 102, 255);
-	private static final Color SPECIAL_ORB_RECHARGE_COLOR = new Color(153, 204, 255, 50);
 
 	private final Client client;
-	private final SpecOrbPlugin plugin;
+	private final MinimapOrb orb;
 	private int lastSpecialPercent = 0;
 	private int tickCounter = 0;
 
 	@Inject
-	public SpecOrbOverlay(Client client, SpecOrbPlugin plugin)
+	public SpecOrbOverlay(Client client, EventBus eventBus)
 	{
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_WIDGETS);
 		this.client = client;
-		this.plugin = plugin;
+		try
+		{
+			BufferedImage icon = ImageIO.read(getClass().getResourceAsStream("special_orb_icon.png"));
+			orb = new MinimapOrb(SPECIAL_ORB_BACKGROUND_COLOR, SPECIAL_ORB_BACKGROUND_COLOR, icon);
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+		eventBus.register(this);
```

Comment:
Generally I prefer for the plugins to be the only thing subscribing to events and then delegating them